Playmate/View Controllers/Home Tab/HomeViewController.m

```diff
@@ -26,187 +26,141 @@ @interface HomeViewController () <UITableViewDelegate, UITableViewDataSource, DZ
+	[super viewDidLoad];
+
+	self.tableView.delegate = self;
+	self.tableView.dataSource = self;
+
+	self.sessionList = [[NSMutableArray alloc] init];
+
+	PFUser *me = [[PFUser currentUser] fetchIfNeeded];
+
+	NSString *greeting;
+	NSDate *now = [NSDate now];
+	if ([now hour] >= 17) {
+		greeting = @"Good Evening, ";
+	} else if ([now hour] >= 12) {
+		greeting = @"Good Afternoon, ";
+	} else {
+		greeting = @"Good Morning, ";
+	}
+
+	self.welcomeLabel.text = [greeting stringByAppendingString:me[@"firstName"][0]];
+
+	// set up refresh control
+	self.refreshControl = [[UIRefreshControl alloc] init];
+    [self.refreshControl addTarget:self
+                         action:@selector(fetchData)
+                         forControlEvents:UIControlEventValueChanged];
     [self.tableView addSubview:self.refreshControl];
 }
 
 - (void)viewDidAppear:(BOOL)animated {
-    [self fetchData];
+	[self fetchData];
 }
 
 - (void)fetchData {
-    PFQuery *query = [PFQuery queryWithClassName:@"SportsSession"];
-    query.limit = 20;
-    
-    [query orderByAscending:@"occursAt"];
-
-    // fetch data asynchronously
-    [query findObjectsInBackgroundWithBlock:^(NSArray *sessions, NSError *error) {
-        if (sessions != nil) {
-            
-            [self.sessionList removeAllObjects];
-            [self filterSessions:sessions];
-            
-            [self.tableView reloadData];
-        } else {
-            NSLog(@"%@", error.localizedDescription);
-        }
-        [self.refreshControl endRefreshing];
-    }];
+	PFQuery *query = [PFQuery queryWithClassName:@"SportsSession"];
+	query.limit = 20;
+
+	[query orderByAscending:@"occursAt"];
+
+	// fetch data asynchronously
+	[query findObjectsInBackgroundWithBlock:^(NSArray *sessions, NSError *error) {
+	         if (sessions != nil) {
+
+			 [self.sessionList removeAllObjects];
+			 [self filterSessions:sessions];
+
+			 [self.tableView reloadData];
+		 } else {
+			 NSLog(@"%@", error.localizedDescription);
+		 }
+	         [self.refreshControl endRefreshing];
+	 }];
 }
 
 // Filter out sessions that do not contain self
 - (void)filterSessions:(NSArray *)sessions {
-    
-    NSMutableArray *tempList = (NSMutableArray *)sessions;
-    PFUser *currUser = [[PFUser currentUser] fetchIfNeeded];
-    
-    for (Session *sesh in tempList) {
-        
-        for (PFUser *user in sesh[@"playersList"]) {
-            [user fetchIfNeeded];
-            
-            NSDate *now = [NSDate date];
-            NSComparisonResult result = [now compare:sesh.occursAt];
-            
-            if ([currUser.username isEqualToString:user.username] && result == NSOrderedAscending) {
-                [self.sessionList addObject:sesh];
-                break;
-            }
-        }
-    }
-}
+	NSMutableArray *tempList = (NSMutableArray *)sessions;
+	PFUser *currUser = [[PFUser currentUser] fetchIfNeeded];
 
-#pragma mark - Empty table view protocol methods
+	for (Session *sesh in tempList) {
```

Comment:
Try to avoid abbreviations. 

I know they are tempting while writting, but after a couple of months when you come back to the code, you will  be glad to have used the entire word.

I would do
`Session *session in sessions`